### PR TITLE
Glut Cipher implementation

### PR DESCRIPTION
**Pull request by nicohasa**
For #3219

### CODE REVIEW

File: test/clj/game_test/macros.clj [hunk 1]
```diff
@@ -6,6 +6,7 @@
   `(let [~'state ~s
          ~'get-corp (fn [] (:corp @~'state))
          ~'get-runner (fn [] (:runner @~'state))
+         ~'get-run (fn [] [:run @~'state])
```
Comment (Saintis) [inline test/clj/game_test/macros.clj]: I think this should be `()` and not `[]`.

File: test/clj/game_test/core.clj [hunk 2]
```diff
@@ -111,6 +111,10 @@
   [state pos]
   (get-in @state [:runner :rig :facedown pos]))
 
+(defn get-discarded
+  [state side pos]
+  (get-in @state [side :discard pos]))
```
Comment (Saintis) [inline test/clj/game_test/core.clj]: I'd add a "default" value for pos as well for when you get the top discarded card.

**Comment by nealterrell** [comment]
Does this  handle the UFAQ rule that using this on Archives with <5 cards skips the access phase entirely?

**Comment by nicohasa** [comment]
Yep, it does. Will fix the other comments soon-ish.

**Pull request merged**